### PR TITLE
Update triggers core interceptor to create secret when necessary

### PR DESCRIPTION
**Pull request by savitaashture**
The core interceptor will create secrets in only 2 cases:
1. when there is no secret present in the cluster.
2. when certificates expire within the existing secret.

If secret exist and have valid certificates no need to recreate secret everytime

Signed-off-by: Savita Ashture <sashture@redhat.com>

Fixes : https://github.com/tektoncd/triggers/issues/1638

# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [ ] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/approve

File: pkg/interceptors/server/server_test.go [hunk 1]
```diff
@@ -236,7 +236,7 @@ func createSecret(t *testing.T, noAfter time.Time) (v1.CoreV1Interface, []byte,
 	if _, err := clientSet.Secrets(namespace).Create(ctx, localObject, metav1.CreateOptions{}); err != nil {
 		t.Error(err)
 	}
-	sCert, caCert, err := createCerts(ctx, clientSet, noAfter, logger.Sugar())
+	sCert, caCert, err := createCerts(ctx, clientSet, noAfter, logger.Sugar(), false)
```
Comment (khrm) [inline pkg/interceptors/server/server_test.go:239]: Don't we need a test for the `true` scenario also?

File: pkg/interceptors/server/server_test.go [hunk 2]
```diff
@@ -236,7 +236,7 @@ func createSecret(t *testing.T, noAfter time.Time) (v1.CoreV1Interface, []byte,
...
 	return clientSet, sCert, caCert, err
```
Comment (khrm) [inline pkg/interceptors/server/server_test.go:240]: Don't we need a test for the true scenario also?

Comment (savitaashture) [reply pkg/interceptors/server/server_test.go:240]: updated test  to work with  true scenario as well

**Comment by khrm** [comment]
/assign @dibyom

/approve

Let's get this merge.

**Pull request merged**